### PR TITLE
Nullify api_backend when APIAP routing policy is used

### PR DESCRIPTION
**Pull request by josemigallas**
**What this PR does / why we need it**:

When a product is using multiple backend, system still injects the old api_backend attribute into the proxy configuration. With this PR, `api_backend` is set to `nil` for new proxy configs when there are more than 1 backend.

**Which issue(s) this PR fixes** 

[THREESCALE-3775: Nullify old api_backend attribute when APIAP routing policy is used](https://issues.jboss.org/browse/THREESCALE-3775)

### CODE REVIEW

**Comment by guicassolato** [comment]
We should check with @3scale/apicast what happens when
- `api_backend` is `null`,
- no routing rule matches, and
- the product includes the "catch-all" mapping rule ("/").

We need to ensure that it returns a 404 instead of an ugly 500.

cc @eloycoto

File: test/unit/services/apicast_v2_deployment_service_test.rb [hunk 1]
```diff
@@ -56,6 +57,13 @@ def test_call
...
 
+  test 'api_backend is nil when product has multiple backends' do
+    @proxy.stubs(with_subpaths?: true)
 
-end
+    config = Service.new(@proxy).call(environment: ProxyConfig::ENVIRONMENTS.first)
+    json_content = JSON.parse(config.content, symbolize_names: true)
+    api_backend = json_content.dig(:proxy, :api_backend)
 
+    assert_equal api_backend, nil
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply test/unit/services/apicast_v2_deployment_service_test.rb]: 🤦‍♂

**Comment by eloycoto** [comment]
quick one, so the APICast config should be like this no? 

```
{
  "services": [
    {
      "id": 200,
      "backend_version": 1,
      "backend_authentication_type": "service_token",
      "backend_authentication_value": "token-value",
      "proxy": {
        "hosts": [
          "one",
          "local-apicast.acalustra.com"
        ],
        "proxy_rules": [
          {
            "last": true,
            "pattern": "/",
            "http_method": "GET",
            "metric_system_name": "hits",
            "delta": 1
          }
        ],
        "policy_chain": [
          {
            "name": "apicast.policy.apicast"
          }
        ]
      }
    }
  ]
}
```
And should return 404, should be 502 no gateway no?

Regards

**Comment by eloycoto** [comment]
Just a note, by default is sending a 200 back: 
https://github.com/3scale/APIcast/blob/b6aaa3bfb281c4132589bff56d576607665ea28c/gateway/src/apicast/policy/apicast/apicast.lua#L104-L108

**Comment by guicassolato** [comment]
@eloycoto, technically a 502 would mean an upstream was contacted by the gateway but failed to respond. I think the response code should be a 404 instead, because actually there is no route the requested resource. Semantics? WDYT?

**Comment by josemigallas** [comment]
@eloycoto @guicassolato
Can we merge this?

**Pull request merged**